solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/util/SolaceErrorMessageHandler.java

```diff
@@ -73,9 +65,11 @@ public void handleMessage(Message<?> message) throws MessagingException {
 		}
 
 		try {
-			AckUtils.reject(acknowledgmentCallback);
+			if (!SolaceAckUtil.republishToErrorQueue(acknowledgmentCallback)) {
+				AckUtils.requeue(acknowledgmentCallback);
+			}
 		} catch (SolaceAcknowledgmentException e) {
-			if (ExceptionUtils.indexOfType(e, SolaceStaleMessageException.class) > -1) {
+			if (ExceptionUtils.indexOfType(e, IllegalStateException.class) > -1) {
```

Comment:
I think we might want to just rethrow this unless we're absolutely sure that this is because of a stale message and this is safe ignore.

Otherwise, we'll accidentally lose messages, because when the error message handler returns successfully, SCSt treats the message as successfully handled and the message is then ACCEPTed.

Reply:
Specifically, in [BasicInboundXMLMessageListener](https://github.com/mayur-solace/solace-spring-cloud/blob/master/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/BasicInboundXMLMessageListener.java#L62-L64), if the error handler function returns successfully (and did throw an exception), then this causes `errorHandlerFunction.apply()` to return `true`, and the message is then auto-acked.

Similar logic applies in the [RetryableInboundXMLMessageListener](https://github.com/mayur-solace/solace-spring-cloud/blob/master/solace-spring-cloud-stream-binder/solace-spring-cloud-stream-binder-core/src/main/java/com/solace/spring/cloud/stream/binder/inbound/RetryableInboundXMLMessageListener.java#L60-L61).